### PR TITLE
backport: bump python 3.9

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Why not

## What was done?
Bump python version

## How Has This Been Tested?
See CI

## Breaking Changes
None

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: .python-version [hunk 1]
```diff
@@ -1 +1 @@
-3.8.16
+3.9.17
```
Comment (knst) [inline .python-version]: missing changes in doc/dependencies.md ?

**Review by knst — COMMENTED** [review]
CI failure seems unrelated see #6315
```
p2p_net_deadlock.py                                | ✖ Failed  | 2 s
```

can you rebase to see CI result?

File: .python-version (hunk 1 above)
Comment (UdjinM6) [reply .python-version]: bump @PastaPastaPasta

**Review by knst — APPROVED** [review]
utACK da45a6743aff2e799300b10e5fdb72916822f298

**Review by kwvg — APPROVED** [review]
utACK da45a6743aff2e799300b10e5fdb72916822f298

**Review by UdjinM6 — APPROVED** [review]
utACK da45a6743aff2e799300b10e5fdb72916822f298

**Pull request merged**